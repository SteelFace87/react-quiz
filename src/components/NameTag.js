import React, { PureComponent } from 'react';
import TextSelector from './TextSelector';
import Shape from './Shape';
import ColorSelector from './ColorSelector';
import FontColorSelector from './FontColorSelector';


export default class TextFormatter extends PureComponent {
  state = {
    text: '',
    message: '',
    color:'purple',
    fontColor:'black'
  }

  handleSubmit = (e) => {
    e.preventDefault();
    this.setState({ message: this.state.text });
    this.setState({ text: '' });

  }
  handleTextChange = ({ target }) => {
    const text = target.value;
    this.setState({ text: text });
  }
  handleColorChange = ({ target }) => {
    const color = target.value;
    this.setState({ color: color });
  }
  handleFontColorChange = ({ target }) => {
    const color = target.value;
    this.setState({ fontColor: color });
  }

  render() {
    const { 
      fontColor,
      text,
      message,
      color } = this.state;

    return (
      <>
        <TextSelector text={text} handleTextChange={this.handleTextChange} handleSubmit={this.handleSubmit} />
        <ColorSelector handleColorChange={this.handleColorChange}/>
        <FontColorSelector handleFontColorChange={this.handleFontColorChange}/>
        <Shape text={message} color={fontColor} backgroundColor={color} />

      </>
    );
  }

}
