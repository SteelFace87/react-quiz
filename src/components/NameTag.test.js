import React from 'react';
import { shallow } from 'enzyme';
import NameTag from './NameTag';

describe('NameTag Component', ()=>{
  it('renders stuff', ()=>{
    const wrapper =  shallow(<NameTag/>);
    expect(wrapper).toMatchSnapshot();
  });
})
;
