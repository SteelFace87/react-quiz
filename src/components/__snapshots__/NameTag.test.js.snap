// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameTag Component renders stuff 1`] = `
<Fragment>
  <TextInputForm
    handleSubmit={[Function]}
    handleTextChange={[Function]}
    text=""
  />
  <ColorOptions
    handleColorChange={[Function]}
  />
  <Shape
    backgroundColor="purple"
    color="purple"
    text=""
  />
</Fragment>
`;
